internal/provider/provider.go

```diff
@@ -77,6 +92,39 @@ func (p *infisicalProvider) Schema(ctx context.Context, _ provider.SchemaRequest
 				Sensitive:   true,
 				Description: "Machine identity client secret. Used to fetch/modify secrets for a given project",
 			},
+			"auth": schema.SingleNestedAttribute{
+				Optional:    true,
+				Description: "The configuration values for authentication",
+				Attributes: map[string]schema.Attribute{
+					"universal": schema.SingleNestedAttribute{
+						Optional:    true,
+						Description: "The configuration values for Universal Auth",
+						Attributes: map[string]schema.Attribute{
+							"client_id": schema.StringAttribute{
```

Comment:
Should `client_id` and `client_secret` be `optional: true`?

Reply:
ah right good catch, they should be Required

Reply:
hmmm for this one, we're currently allowing users to define client_id and client_secret via ENV rather than directly in the configuration which is why this has to be optional